x-pack/plugins/fleet/server/services/agents/upgrade_action_runner.ts

```diff
@@ -167,6 +169,8 @@ export async function upgradeBatch(
 
   const actionId = options.actionId ?? uuidv4();
   const total = options.total ?? givenAgents.length;
+  const currentNameSpace = getCurrentNamespace(soClient);
```

Comment:
it looks like this code could also run in a background task in that case the soClient namespace will not be correct (will be default),
we may want to use the agent namespace instead to populate that field

Reply:
Thanks for flagging this up! I've made a change (pass space if to batch function), let me know if it looks ok.

Reply:
Thinking more about this: in the current state, it looks like `spaceId` might be `undefined` in case this runs as a background task. A solution could be to modify the action runner to take a space id.

I'm not sure about using the `namespaces` property of (of the) agents: could there be a scenario where we have agents belonging to multiple spaces? e.g. agent1 in spaces A and B, agent 2 in spaces B and C. In that case, we couldn't use agents to assign a space to the action.

Reply:
I think as the task will come from an API we will probably have a namespace in the API request  that we could pass to the task and use as a parameter  for the action runner, does that make sense to you?

Reply:
I thought this would affect the case when an action runner is used, for example for bulk reassign when the number of agents exceeds the batch size:
https://github.com/elastic/kibana/blob/813025c04bb0dcdac3a117416ff159932c11edc0/x-pack/plugins/fleet/server/services/agents/reassign.ts#L128-L138

I should test this properly, but I expect that in this case the current solution will not provide a namespace. Do you mean that the API request should pass it (or already is)?

Reply:
I think something like this should work
```
{ 
     ...options, 
spaceId: soClient.getCurrentNamespace(),
     batchSize, 
     total: res.total, 
     newAgentPolicyId, 
   }, 
```